### ISSUE TITLE
boolean-prop-naming is missing default config

### ISSUE THREAD

**Issue by avaly** [yannickcr/eslint-plugin-react#1551]
It seems that the `boolean-prop-naming` rule does not enforce any default config (especially for `rule` as the docs mention [here](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/boolean-prop-naming.md#rule))

Sample code:

```js
import * as React from 'react';
import PropTypes from 'prop-types';

class ReactClassComponent extends React.Component {
	static propTypes = {
		error: PropTypes.bool,
		hasError: PropTypes.bool,
		isOptional: PropTypes.bool,
		optional: PropTypes.bool,
	};

	render() {
		return null;
	}
}
```

Sample eslint configuration:

```js
	plugins: ['react'],
	rules: {
		'react/boolean-prop-naming': ['error'],
	},
```

Running `eslint .` gives no errors.

However switching the rule config to `['error', { rule: '^(is|has)[A-Z]([A-Za-z0-9]?)+' }]` will throw linting error for the sample code above.

**Comment by jseminck** [comment]
The default is definitely configured inside the code-base: https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/rules/boolean-prop-naming.js#L35

What ESLint version are you using?

I'll try to reproduce this locally.

**Comment by jomasti** [comment]
It looks like that `default` field is only metadata, and it doesn't appear to be actually used when parsing the rule schema. The tests are missing an example with no options. It will indeed not work without passing the rule in the options.

**Comment by jseminck** [comment]
Yep. It does not work. It also does not seem to work when using `PropTypes.bool.isRequired` (adding `.isRequired`).

**Comment by ljharb** [comment]
Wow, that actually seems like it's a bug in eslint core; either "default" should be respected or it should throw.

Regardless, it's also a bug in this rule that we're lacking these tests.

**Comment by jseminck** [comment]
Asked the question in ESLint Gitter chat regarding the default value. Otherwise will make a PR to add it from our side, and would also like to make sure that `.isRequired` works here (different issue though).

**Comment by JamyGolden** [comment]
@jseminck this still seems to be a bug. Do you have any idea if there has been any progress with the issue?

**Comment by ljharb** [comment]
I think that this fix will also apply to #1540.
